ncm-network/src/main/pan/components/network/types/network/backend/nmstate.pan

```diff
@@ -5,16 +5,31 @@ declaration template components/network/types/network/backend/nmstate;
 @documentation{
     NetworkManager device configuration for drop in config file.
 }
+
+type structure_nm_main_config = {
+    @{Set the DNS processing mode for NetworkManager}
+    "dns" : choice("dnsmasq", "default", "none", "systemd-resolved")
+    @{Specify devices for which NetworkManager should not create default wired connection}
+    "no-auto-default" ? string
```

Comment:
Should this be a list of strings instead?

Reply:
yes probably, will require a bit more thought on this as it can be interface names or mac addresses etc, 
I'm not sure who will need this right now, we don't, just added it as bonus, so I'm inclined not to include this in this PR, and look into later, perhaps, if need arise?